src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -819,6 +782,15 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
                 kind = TypedConstantKind.Primitive;
                 defaultValue = ((ContextualAttributeBinder)this).AttributedMember.GetMemberCallerName();
             }
+            else if (!IsEarlyAttributeBinder && syntax.ArgumentList is not null &&
+                getCallerArgumentArgumentIndex(parameter, argumentsToParams) is int argumentIndex && argumentIndex > -1 && argumentIndex < argumentsCount)
+            {
+                Debug.Assert(argumentsCount <= syntax.ArgumentList.Arguments.Count);
+                CheckFeatureAvailability(syntax.ArgumentList, MessageID.IDS_FeatureCallerArgumentExpression, diagnostics);
+                parameterType = Compilation.GetSpecialType(SpecialType.System_String);
```

Comment:
>parameterType = Compilation.GetSpecialType(SpecialType.System_String); [](http://example.com/codeflow?start=16&length=70)

Is there a guarantee that the parameter type is a string? If so, why not pull it from the ParameterSymbol and assert? If there is no guarantee, we should have a test covering the scenario.

Reply:
I followed the same pattern used by other caller member info above. But I'll look into that.

Reply:
@AlekseyTs It's not guaranteed to be a string. It can be an object, and in such case if we pulled it from the ParameterSymbol, emit will fail with `Binary format of the specified custom attribute was invalid.`
I'll add a test covering this.

Reply:
@AlekseyTs The diagnostic is reported elsewhere. This is not done in binder so that the error is still produced even if there is no usage of the method having the argument expression parameter.

Reply:
>The diagnostic is reported elsewhere. This is not done in binder so that the error is still produced even if there is no usage of the method having the argument expression parameter (i.e, the diagnostic is produced at the declaration, not the call site).

I think we are talking about different diagnostics. I am talking about "bad" string type. It might not be part of the signature. The error is not about missing conversion.

Reply:
@AlekseyTs I'm not sure I understand this (sorry for that). Does this suggest to use the following helper?

https://github.com/dotnet/roslyn/blob/f6ba23211d9272426214fe055ab8f1e9983e6dd4/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs#L1468

Reply:
> Does this suggest to use the following helper?

>[https://github.com/dotnet/roslyn/blob/f6ba23211d9272426214fe055ab8f1e9983e6dd4/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs#L1468](https://github.com/dotnet/roslyn/blob/f6ba23211d9272426214fe055ab8f1e9983e6dd4/src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs#L1468)

Yes. Binder.GetSpecialType helper that takes diagnostics